druid/src/widget/list.rs

```diff
@@ -45,6 +165,41 @@ pub struct List<T> {
 impl<T: Data> List<T> {
     /// Create a new list widget. Closure will be called every time when a new child
     /// needs to be constructed.
+    ///
+    /// The list widget can have a function passed in as a closure, or an abstract closure can also
+    /// be provided.
+    ///
+    /// # Example
+    /// ```
+    /// # use druid::widget::{ Label, List };
+    /// # use druid::{ Data, Lens, Widget, WidgetExt, Env};
+    /// # use druid::im::Vector;
+    /// # #[derive(Clone, Data, Lens)]
+    /// # struct AppState {
+    /// #     list_data: Vector<String>,
+    /// # }
+    /// fn root() -> impl Widget<AppState> {
+    ///     let list = List::new(list_item)
+    ///         .lens(AppState::list_data);
+    ///     list
+    /// }
+    ///
+    /// fn list_item() -> impl Widget<String> {
+    ///     let label = Label::new(|data: &String, _env: &Env|
+    ///         data.clone());
+    ///     label
+    /// }
+    ///
+    /// //This widget is the same as the two widgets above
+    /// //combined.
+    /// fn combined() -> impl Widget<AppState> {
+    ///     let list = List::new(||
+    ///             Label::new(|data: &String, _env: &Env|
+    ///                 data.clone()))
+    ///         .lens(AppState::list_data);
+    ///     list
+    /// }
```

Comment:
You can probably remove the `combined` function; the other two are descriptive enough on their own.